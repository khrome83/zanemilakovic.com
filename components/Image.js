// @ts-check --jsx

import PropTypes from 'prop-types';
import Head from 'next/head';

// Picture Element
const Picture = ({ children }) => (
  <picture>
    { children }
    <Head>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/picturefill/3.0.3/picturefill.min.js" integrity="sha256-iT+n/otuaeKCgxnASny7bxKeqCDbaV1M7VdX1ZRQtqg=" crossOrigin="anonymous" />
    </Head>
    <style jsx>{`
      .header {
        width: 100%;
      }
    `}</style>
  </picture>
);

Picture.propTypes = {
  children: PropTypes.node.isRequired,
};

// Image Element
const Image = props => (
  <img {...props} alt={props.alt} />
);

Image.defaultProps = {
  sizes: null,
  srcSet: null,
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  srcSet: PropTypes.string,
  sizes: PropTypes.string,
};

// Source Element
const Source = props => (
  <source {...props} />
);

Source.defaultProps = {
  sizes: null,
};

Source.propTypes = {
  srcSet: PropTypes.string.isRequired,
  media: PropTypes.string.isRequired,
  sizes: PropTypes.string,
};

export { Image as default, Picture, Source };
