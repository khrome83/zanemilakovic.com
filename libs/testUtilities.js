// @ts-check --jsx
/* eslint-disable no-param-reassign */

const cleanSnapshots = (node) => {
  for (const prop in node) {
    if (prop === 'className') {
      const className = node[prop].replace(/jsx-[0-9]+[\s]?/g, '');
      node.className = className || '';
      if (!node.className) {
        delete node.className;
      }
    } else if (typeof node[prop] === 'object') {
      cleanSnapshots(node[prop]);
    }
  }
};

const lib = {
  cleanSnapshots,
};

export { lib as default, cleanSnapshots };
