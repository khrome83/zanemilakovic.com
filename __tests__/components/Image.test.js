import React from 'react';
import renderer from 'react-test-renderer';
import { cleanSnapshots } from '../../libs/testUtilities';
import Image, { Picture, Source } from '../../components/Image';

test('Image renders correctly', () => {
  const props = {
    sizes: '(max-width: 1982px) 100vw, 1982px',
    srcSet: `
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_768.jpg 768w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1096.jpg 1096w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1352.jpg 1352w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1597.jpg 1597w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1830.jpg 1830w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg 1982w"
    `,
    src: 'kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg',
    alt: 'Laptop sitting on desk',
  };

  const component = renderer.create(<Image {...props} />);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('Source renders correctly', () => {
  const props = {
    media: '(max-width: 767px)',
    sizes: '(max-width: 1534px) 100vw, 1534px',
    srcSet: `
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_320.jpg 320w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_733.jpg 733w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1014.jpg 1014w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1293.jpg 1293w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1499.jpg 1499w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1534.jpg 1534w
    `,
  };

  const component = renderer.create(<Source {...props} />);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('Picture renders correctly', () => {
  const sourceProps = {
    media: '(max-width: 767px)',
    sizes: '(max-width: 1534px) 100vw, 1534px',
    srcSet: `
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_320.jpg 320w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_733.jpg 733w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1014.jpg 1014w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1293.jpg 1293w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1499.jpg 1499w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1534.jpg 1534w
    `,
  };

  const imageProps = {
    sizes: '(max-width: 1982px) 100vw, 1982px',
    srcSet: `
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_768.jpg 768w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1096.jpg 1096w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1352.jpg 1352w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1597.jpg 1597w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1830.jpg 1830w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg 1982w"
    `,
    src: 'kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg',
    alt: 'Laptop sitting on desk',
  };

  const component = renderer.create(
    <Picture>
      <Source {...sourceProps} />
      <Image {...imageProps} />
    </Picture>
  );
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});
