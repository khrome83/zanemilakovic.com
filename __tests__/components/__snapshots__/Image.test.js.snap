// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<img
  alt="Laptop sitting on desk"
  sizes="(max-width: 1982px) 100vw, 1982px"
  src="kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg"
  srcSet="
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_768.jpg 768w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1096.jpg 1096w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1352.jpg 1352w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1597.jpg 1597w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1830.jpg 1830w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg 1982w\\"
    "
/>
`;

exports[`Picture renders correctly 1`] = `
<picture>
  <source
    media="(max-width: 767px)"
    sizes="(max-width: 1534px) 100vw, 1534px"
    srcSet="
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_320.jpg 320w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_733.jpg 733w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1014.jpg 1014w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1293.jpg 1293w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1499.jpg 1499w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1534.jpg 1534w
    "
  />
  <img
    alt="Laptop sitting on desk"
    sizes="(max-width: 1982px) 100vw, 1982px"
    src="kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg"
    srcSet="
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_768.jpg 768w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1096.jpg 1096w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1352.jpg 1352w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1597.jpg 1597w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1830.jpg 1830w,
      kari-shea-272383_bbmubb_ar_4_3,c_fill,g_auto__c_scale,w_1982.jpg 1982w\\"
    "
  />
</picture>
`;

exports[`Source renders correctly 1`] = `
<source
  media="(max-width: 767px)"
  sizes="(max-width: 1534px) 100vw, 1534px"
  srcSet="
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_320.jpg 320w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_733.jpg 733w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1014.jpg 1014w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1293.jpg 1293w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1499.jpg 1499w,
      kari-shea-272383_bbmubb_ar_1_1,c_fill,g_auto__c_scale,w_1534.jpg 1534w
    "
/>
`;
